cps/kobo.py

```diff
@@ -392,6 +394,222 @@ def get_metadata(book):
     return metadata
 
 
+@kobo.route("/v1/library/tags", methods=["POST"])
+@login_required
+# Creates a Shelf with the given items, and returns the shelf's uuid.
+def HandleTagCreate():
+    shelf_request = request.json
+    name, items = None, None
+    try:
+        name = shelf_request["Name"]
+        items = shelf_request["Items"]
+    except KeyError:
+        log.debug("Received malformed v1/library/tags request.")
+        abort(400, description="Malformed tags POST request. Data is missing 'Name' or 'Items' field")
+
+    shelf = ub.session.query(ub.Shelf).filter(and_(ub.Shelf.name) == name, ub.Shelf.user_id ==
```

Comment:
Big little problem, the name is not/no longer unique per user:
![grafik](https://user-images.githubusercontent.com/6438895/79682263-f9d0da80-8220-11ea-82e1-94daa0b5dd25.png)
Every user can generate a public and private shelf with the same name, so we should check it during loading shelf name. I think we should prefer the private shelf over the public one with the same name. ( We are having 2 cases to consider 
1) name is not valid or not belonging to user
2) name is unique private or unique public shelf owned by user
3) name is used for private and public shelf of user
This problem occurs for ever shelf request with filtering from name

Reply:
Another idea would be: In the UI to the public shelfs the text 'public' is appended, so maybe doing this here would also be possible. But this text is translatable, so it depends on the user interface language.

Reply:
I might be misunderstanding, but I don't think this is an issue for us.

We only use the shelf name during shelf creation. Other API calls (such as adding/removing books from a shelf, or deleting a shelf) rely on the ub.Shelf.uuid field to identify the shelf they operate on.
The only reason we check the shelf name here is to prevent the user from **creating** multiple shelves with the same name. However if shelves with the same name do exist, that won't affect the Kobo sync protocol in any way.